Add new endpoint reference page for the validators module (#1811)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/api/module-rpc-api/validators-endpoints.adoc

= Validators module: Endpoint reference
Mona Bärenfänger <mona@lightcurve.io> Chris Braithwaite <christopher.braithwaite@lightcurve.io>
// Settings
:toc: preamble

//URLs

> :url_typedoc_auth: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorModule.html

Reviewer: ```suggestion
:url_typedoc_auth: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorsModule.html
```

//Project URLs
:url_integrate_endpoints: integrate-blockchain/connecting-endpoints.adoc

This page summarizes all available API endpoints of the {url_typedoc_auth}[Validators module^].

TIP: Endpoints can be invoked as explained in the xref:{url_integrate_endpoints}[] guide.

> == `ValidatorBLSKey`

Reviewer: The names of endpoints should have the namespace as prefix as well.

This endpoint is used to retrieve the BLS public key of a specific validator in the network.

.Request
[tabs]
=====
Specification::
+
--
.Parameters

> [cols="1,1,2,2",options="header",stripes="hover"]

Reviewer: ```suggestion
[cols="1,1,4,4",options="header",stripes="hover"]
```

|===
|Name
|Type
|Description
|Sample

|`proofOfPossession`
|string
|Validates and verifies ownership of tokens.
|b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f

|`blsKey`
|string
|blsKey in hex format.
|83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa
|===

--
Node CLI::
+
--
[source,bash]
----

> lisk-core endpoint:invoke validateBLSKey '{ "blsKey": "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa", "ProofOfPossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty

Reviewer: The endpoint name should have the namespace prefix.

----

--
cURL::
+
--
[source,bash]
----
curl --location --request POST 'http://localhost:7887/rpc' \
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "validator_validateBLSKey",
    "params": {
         "blsKey":  "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa", "proofOfpossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty
    }
}'
----
--
=====

.Response
[%collapsible]
====
.Example output
[source,json]
----
{
  "valid": true
}
----

If there is no BLS key in the store it would return as `false`.
====

== `getValidator`
Returns the `generatorKey` and the `blsKey` of the validator module.

.Request
[tabs]
=====
Specification::
+
--
.Parameters

> [cols="1,1,2,2",options="header",stripes="hover"]

Reviewer: I would increase the column width to make more space for the lengthy content.
```suggestion
[cols="1,1,4,4",options="header",stripes="hover"]
```

|===
|Name
|Type
|Description
|Sample

|`address`
|string
|The Lisk32 address to query.
|lske5sqed53fdcs4m9et28f2k7u9fk6hno9bauday

--
Node CLI::
+
--
[source,bash]
----

> lisk-core endpoint:invoke getValidator '{"address":"lske5sqed53fdcs4m9et28f2k7u9fk6hno9bauday"}' --pretty

Reviewer: Same issue as mentioned earlier about prefix!

= Validators module: Endpoint reference
Mona Bärenfänger <mona@lightcurve.io> Chris Braithwaite <christopher.braithwaite@lightcurve.io>
// Settings
:toc: preamble

//URLs
:url_typedoc_auth: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorsModule.html
//Project URLs
:url_integrate_endpoints: integrate-blockchain/connecting-endpoints.adoc

This page summarizes all available API endpoints of the {url_typedoc_auth}[Validators module^].

TIP: Endpoints can be invoked as explained in the xref:{url_integrate_endpoints}[] guide.

== `validators_validateBLSKey`
This endpoint is used to retrieve the BLS public key of a specific validator in the network.

.Request
[tabs]
=====
Specification::
+
--
.Parameters
[cols="1,1,4,4",options="header",stripes="hover"]
|===
|Name
|Type
|Description
|Sample

|`proofOfPossession`
|string

> |Validates and verifies ownership of tokens.

Reviewer 2: Maybe a better description would be, `Proof that the owner of a key pair actually has the private key associated with the public key`

Author: Sounds good..now updated accordingly 👍

|b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f

|`blsKey`
|string
|blsKey in hex format.
|83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa
|===

--
Node CLI::
+
--
[source,bash]
----

> lisk-core endpoint:invoke validators_validateBLSKey '{ "blsKey": "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa", "ProofOfPossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty

Reviewer 2: ```suggestion
lisk-core endpoint:invoke validators_validateBLSKey '{ "blsKey": "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa", "proofOfPossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty
```

= Validators module: Endpoint reference
Mona Bärenfänger <mona@lightcurve.io> Chris Braithwaite <christopher.braithwaite@lightcurve.io>
// Settings
:toc: preamble

//URLs
:url_typedoc_auth: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorsModule.html
//Project URLs
:url_integrate_endpoints: integrate-blockchain/connecting-endpoints.adoc

This page summarizes all available API endpoints of the {url_typedoc_auth}[Validators module^].

TIP: Endpoints can be invoked as explained in the xref:{url_integrate_endpoints}[] guide.

== `validators_validateBLSKey`

> This function checks that the input BLS key `blsKey` has not been registered before in the chain and that the proof of possession, `proofOfPossession` for the BLS key is valid.
> // This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Reviewer: ```suggestion
The API checks whether the specified `blsKey` is registered on the chain or not and whether the `proofOfPossession` for the BLS key is valid.
The API will return `false` if the `blsKey` is already registered.
```

|===

--
Node CLI::
+
--
[source,bash]
----
lisk-core endpoint:invoke validators_validateBLSKey '{ "blsKey": "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa", "proofOfPossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty
----

--
cURL::
+
--
[source,bash]
----
curl --location --request POST 'http://localhost:7887/rpc' \
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "validators_validateBLSKey",
    "params": {
         "blsKey":  "83d4f8300dda93e62b375264e8e47bc224cbb8f2ecf4c2e848ac9758d6d8fd263c22ef2d9be18b066b8efcab80d6bdaa",
         "proofOfPossession": "b92b11d66348e197c62d14af1453620d550c21d59ce572d95a03f0eaa0d0d195efbb2f2fd1577dc1a04ecdb453065d9d168ce7648bc5328e5ea47bb07d3ce6fd75f35ee51064a9903da8b90f7dc8ab4f2549b834cb5911b883097133f" }' --pretty
    }
}'
----
--
=====

.Response
[%collapsible]
====
.Example output
[source,json]
----
{
  "valid": true
}
----

> If there is no BLS key in the store it would return as `false`.

Reviewer: I think this might be the other way around as far as the [lip](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#validateblskey) is concerned.
``` suggestion
If there is no BLS key in the store it would return as `true`. If the input passed to the API is invalid, or the `blsKey` already exist, the response will be `false`.
```

:toc: preamble

//URLs
:url_typedoc_auth: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorsModule.html
//Project URLs
:url_integrate_endpoints: integrate-blockchain/connecting-endpoints.adoc

This page summarizes all available API endpoints of the {url_typedoc_auth}[Validators module^].

TIP: Endpoints can be invoked as explained in the xref:{url_integrate_endpoints}[] guide.

== `validators_validateBLSKey`

> This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Author: @ishantiw Could you please confirm this is the correct description of the endpoint?

> This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Reviewer 2: You can use the description from LIP https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#validateblskey

> This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Author: Now done 👍

> This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Reviewer: @chris529  The endpoints should always be named as `namespace_nameOfEndpoint `which in this case will be `validators_validateBLSKey`. This is the pattern we follow throughout docs for documenting APIs. Please undo this change.

I think Ishan's link was regarding the description of the endpoint rather than the name of the endpoint.

> This endpoint is used to retrieve the BLS public key of a specific validator in the network.

Author: @TalhaMaliktz OK, yes, I thought the naming convention was the namespace_nameOfEndpoint. Apologies, my misunderstanding 👍
